### PR TITLE
Conn 1921 - fileUtils and ExchangeInfo

### PR DESCRIPTION
**Pull request by trantang**
modifed FileUtils to exchangeInfo
add JUnit for FileUtils

### CODE REVIEW

**Review by ptambellini01 — APPROVED** [review]

File: Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java [hunk 1]
```diff
@@ -791,5 +589,247 @@ public static void copyFolder(File src, File dest, Logger log) throws IOExceptio
...
+    }
+
+    private static Element getEndpointConfigurationVersionByLatest(NodeList configurationList, Logger log) {
+        Element latestVersion = null;
+        Float valueVersion = 0.0F;
+        for (int confIndex = 0; confIndex < configurationList.getLength(); confIndex++) {
+            Element epConfiguration = (Element) configurationList.item(confIndex);
+            String currentVersionString = getTextContentOf(epConfiguration, "version");
+
+            if (currentVersionString.contains("LEVEL")) {
```
Comment (TabassumJafri) [inline Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java]: Needs equalIgnoreCase comparison here

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java [hunk 2]
```diff
@@ -71,7 +74,7 @@
      */
     public static String readFile(String fileName, Logger log) {
         try {
-            log.info("read file: " + fileName);
+            log.info(MessageFormat.format("read file: {0}", fileName));
```
Comment (mhpnguyen) [inline Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java:77]: No need to use MessageFormat.format. You can simple use log.info("read file {0}", fileName);  This applies to all comments below

File: Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java [hunk 3]
```diff
@@ -791,5 +589,247 @@ public static void copyFolder(File src, File dest, Logger log) throws IOExceptio
...
+        String serviceUrl, String matchEndpointVersion) {
+        boolean serviceNodeFound = false;
+        NodeList endpointList = getElementsByTagNameOf(log, organization, "endpointList", "endpoint");
+
+        if (endpointList != null) {
+            for (int epIndex = 0; epIndex < endpointList.getLength(); epIndex++) {
+                Element endpoint = (Element) endpointList.item(epIndex);
+                String endpointName = getFirstElementOf(endpoint, "name").getTextContent();
+
+                if (endpointName.equals(serviceName)) {
```
Comment (mhpnguyen) [inline Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java]: Use ignorecase

File: Product/SoapUI_Test/FileUtils/src/test/java/nhinc/FileUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+public class FileUtilsTest {
+    private final Logger log = mock(Logger.class);
+    private static final String RESOURCE_FOLDER = "src/test/resources/FileUtils";
+    private static final String BACKUP_FOLDER = "src/test/resources/FileUtils/backupHere";
+    private static final String EXCHANGE_FILE = "exchangeInfo.xml";
+    private static final String EXCHANGE_FILE_ORG = "exchangeInfoOrg.xml";
+    private static final String TEST_FILE = "testFile.properties";
+    private static final String TEST_FILE_ORG = "testFileOrg.properties";
+
+    @BeforeClass
```
Comment (mhpnguyen) [inline Product/SoapUI_Test/FileUtils/src/test/java/nhinc/FileUtilsTest.java]: Remove them if we don't use or have any additional logic

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

File: Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java (hunk 2 above)
Comment (trantang) [reply Product/SoapUI_Test/FileUtils/src/main/java/nhinc/FileUtils.java:77]: currently the version: org.apache.log4j.Logger used in the FileUtils; don't have the capability of formatting string

File: Product/SoapUI_Test/FileUtils/src/test/java/nhinc/FileUtilsTest.java (hunk 4 above)
Comment (trantang) [reply Product/SoapUI_Test/FileUtils/src/test/java/nhinc/FileUtilsTest.java]: currently i don't wanted the testFileOrg.properties to be modified; this get flagged as a modified files when we re-compiled the projects; so i copied the testFileOrg.properties to testFile.properties modified the testFile.properties and delete the testFile.properties

**Review by TabassumJafri — APPROVED** [review]

**Review by mhpnguyen — APPROVED** [review]

**Pull request merged**